rule/string-of-int.go

```diff
@@ -0,0 +1,95 @@
+package rule
+
+import (
+	"go/ast"
+	"go/types"
+
+	"github.com/mgechev/revive/lint"
+)
+
+// StringOfIntRule warns when logic expressions contains Boolean literals.
+type StringOfIntRule struct{}
+
+// Apply applies the rule to given file.
+func (r *StringOfIntRule) Apply(file *lint.File, _ lint.Arguments) []lint.Failure {
+	var failures []lint.Failure
+
+	onFailure := func(failure lint.Failure) {
+		failures = append(failures, failure)
+	}
+
+	astFile := file.AST
+	file.Pkg.TypeCheck()
+
+	w := &lintStringInt{file, onFailure}
+	ast.Walk(w, astFile)
+
+	return failures
+}
+
+// Name returns the rule name.
+func (r *StringOfIntRule) Name() string {
+	return "string-of-int"
+}
+
+type lintStringInt struct {
+	file      *lint.File
+	onFailure func(lint.Failure)
+}
+
+func (w *lintStringInt) Visit(node ast.Node) ast.Visitor {
+	ce, ok := node.(*ast.CallExpr)
+	if !ok {
+		return w
+	}
+
+	if !w.isCallStringCast(ce.Fun) {
+		return w
+	}
+
+	if !w.isIntExpression(ce.Args) {
+		return w
+	}
+
+	w.onFailure(lint.Failure{
+		Confidence: 1,
```

Comment:
Maybe we can drop the confidence here? There's still a chance they want to get the codepoint.